Makefile

```diff
@@ -49,6 +49,9 @@ validate-style:
 	diff $(CHANGES_BEFORE) $(CHANGES_AFTER)
 	-rm $(CHANGES_BEFORE) $(CHANGES_AFTER)
 
+validate-python-version:
+	$(PYTHON) -c 'import sys; v=sys.version_info; assert v >= (3, 9), f"Python 3.9 or newer is required. Current version: {v[0]}.{v[1]}.{v[2]}."'
```

Comment:
Will it really break if the version is lower? I don't think it's a good idea to hard fail if it most probably works but we don't test it.

Reply:
That's the idea, indeed. When it fails, it `exit(1)`.

Now in this PR, we do test in the CI.

About the "dropping versions", we were saying "3.5+" but I know for sure it was already not 3.5 compatible. For example, in `conftest.py`, we do `from __future__ import annotations`, which is 3.7+.

So, we're not necessarily dropping versions. Now, we have a real support scheme, and I think that's a better thing fur users (more reliable).

Reply:
I would not like it as a user, if I use old python version and it works even without guarantees I prefer it that way in some cases instead of artificial failing with unsupported version, when we know it will most probably work.There is no reason to hard fail. We can document what we support, but I don't think we should have this check unless it's really needed and is justifiable.

Reply:
This prevents us from evolving a pre-alpha 0.0.6 product.
If we do evolve it without "artificially breaking", that means we'll break it later without any warning (as these are untested versions).

As a user, I'd rather have something that I can rely on. If I'm being told that 0.0.7 is CPython 3.9+ only, then I can still stay on 0.0.6 or fork it myself. But I won't have guarantees that it'll work anymore.

Last but not least, we declared it as a pre-alpha version, we're free to break compatibility anytime (and that's what was confirmed in yesterday's Daily).

As a conclusion, between "we don't have any clear support scheme, try it and it might work this time but maybe not next update" and "okay, it's tested and supported on versions x, y, and z", I'd always put my money on the latter one, as a user.

Also: old EOL'd Python versions do not receive security updates. They're unsafe and should be avoided.

Reply:
@alexole I don't think it's worth debating over, for now. I removed the mandatory checks, just like you requested. And I left the target, so that users can still run it if they want. I hope this compromise is acceptable.

Also, I split the commits in a better way, and removed some deprecated parameters (for isort) that were spewing warnings. Hopefully, we can merge this before the end of the day ;-).

Reply:
Why do you need to keep this target here still? Supported version is mentioned in `setup.py`